tools/src/main/java/com/nedap/archie/json/flat/FlatJsonGenerator.java

```diff
@@ -244,18 +241,18 @@ private void addAttribute(Map<String, Object> result, String pathSoFar, OpenEHRB
             }
         } else if (child instanceof Collection) {
 
-            Map<String, Integer> amountsPerNodeId = new HashMap<>();
-            for(Object c: (Collection<?>) child) {
+            int numberOfNonLocatables = 1; //1-based, sorry
+
+            Integer numberOfPreviousOccurrences = null;
+
+            for (Object c : (Collection<?>) child) {
 
-                int numberOfNonLocatables = 1; //1-based, sory
                 String archetypeNodeId = modelInfoLookup.getArchetypeNodeIdFromRMObject(c);
-                if(archetypeNodeId != null) {
-                    Integer numberOfPreviousOccurrences = amountsPerNodeId.get(archetypeNodeId);
+                if (archetypeNodeId != null) {
                     addAttribute(result, pathSoFar, parent, c, attributeName, numberOfPreviousOccurrences, cAttribute);
-                    numberOfPreviousOccurrences = numberOfPreviousOccurrences == null ? 1: numberOfPreviousOccurrences + 1;
-                    amountsPerNodeId.put(archetypeNodeId, numberOfPreviousOccurrences);
+                    numberOfPreviousOccurrences = numberOfPreviousOccurrences == null ? 1 : numberOfPreviousOccurrences + 1;
                 } else {
-                    addAttribute(result, pathSoFar, parent, c, attributeName, numberOfNonLocatables == 1 ? null: numberOfNonLocatables, cAttribute);
+                    addAttribute(result, pathSoFar, parent, c, attributeName, numberOfNonLocatables == 1 ? null : numberOfNonLocatables, cAttribute);
                     numberOfNonLocatables++;
                 }
             }
```

Comment:
For Nedap I would change this whole part to:
```java
            int collectionIndex = 1; //1-based, sorry
            for (Object c : (Collection<?>) child) {
                addAttribute(result, pathSoFar, parent, c, attributeName, collectionIndex, cAttribute);
                collectionIndex++;
            }
```
So always include indices, starting at 1 and no differentiation between different node ids and non locatables. Maybe make an exception when there is only one element in the collection. This would match the behaviour of UniqueNodePathBuilder.

I have tested some round trips with some test data and it seems to work correctly.



But for other formats this might no be correct.

@pieterbos Would this work for other formats as well or do we make a differentiation in this behaviour?

Reply:
maybe we should differentiate between the previous behaviour, and the Nedap one. Then we can easily release this as a non-breaking bug-fix release.

Reply:
I have created a new PR based on the master branch including this suggestion: #405